src/install/hosted_git_info.zig

```diff
@@ -0,0 +1,601 @@
+    defer as_slice.deinit();
+    const hostname = bun.strings.withoutPrefixComptime(as_slice.slice(), "www.");
+    if (findHostInfoByDomain(hostname)) |host_info| {
+        return host_info;
+    }
+
+    return null;
+}
+
+/// Test whether the given node-package-arg string is a GitHub shorthand.
+///
+/// This mirrors the implementation of hosted-git-info, though it is significantly faster.
+fn isGithubShorthand(
+    npa_str: []const u8,
+) bool {
+    // The implementation in hosted-git-info is a multi-pass algorithm. We've opted to implement a
+    // single-pass algorithm for better performance.
+    //
+    // This could be even faster with SIMD but this is probably good enough for now.
+    if (npa_str.len < 1) {
+        return false;
+    }
+
+    // Implements doesNotStartWithDot
+    if (npa_str[0] == '.' or npa_str[0] == '/') {
+        return false;
+    }
+
+    var pound_idx: ?u32 = null;
+    var seen_slash = false;
+
+    for (npa_str, 0..) |c, i| {
+        switch (c) {
+            // Implement atOnlyAfterHash and colonOnlyAfterHash
+            ':', '@' => {
+                if (pound_idx == null) {
+                    return false;
+                }
+            },
+
+            '#' => {
+                pound_idx = @intCast(i);
+            },
+            '/' => {
+                // Implements secondSlashOnlyAfterHash
+                if (seen_slash and pound_idx == null) {
+                    return false;
+                }
+
+                seen_slash = true;
+            },
+            else => {
+                // Implement spaceOnlyAfterHash
+                if (std.ascii.isWhitespace(c) and pound_idx == null) {
+                    return false;
+                }
+            },
+        }
+    }
+
+    // Implements doesNotEndWithSlash
+    const does_not_end_with_slash =
+        if (pound_idx) |pi|
+            npa_str.len > 2 and npa_str[pi - 1] != '/'
+        else
+            npa_str[npa_str.len - 1] != '/';
+
+    // Implement hasSlash
+    return seen_slash and does_not_end_with_slash;
+}
+
+const UrlProtocolPair = struct {
+    url: []u8,
+    protocol: union(enum) {
+        well_formed: UrlProtocol,
+
+        // A protocol which is not known by the library. Includes the : character, but not the
+        // double-slash, so `foo://bar` would yield `foo:`.
+        custom: []u8,
+
+        // Either no protocol was specified or the library couldn't figure it out.
+        unknown: void,
+    },
+
+    /// Given a protocol pair, create a bun.jsc.URL if possible. May allocate, but owns its memory.
+    pub fn toUrl(self: *const UrlProtocolPair, allocator: std.mem.Allocator) ?*bun.jsc.URL {
+        // Ehhh.. Old IE's max path length was 2K so let's just use that. I searched for a
+        // statistical distribution of URL lengths and found nothing.
+        const LONG_URL_THRESH = 2048;
+
+        var alloc = std.heap.stackFallback(LONG_URL_THRESH, allocator);
```

Comment:
```suggestion
        const long_url_thresh = 2048;

        var alloc = std.heap.stackFallback(long_url_thresh, allocator);
```

Zig doesn't have a different case convention for compile-time constant variables; see e.g. `std.mem.byte_size_in_bits`.